internals-js/src/definitions.ts

```diff
@@ -2516,7 +2530,7 @@ export class Directive<
   }
 
   private isAttachedToSchemaElement(): boolean {
-    return this.parent !== undefined && this.parent instanceof SchemaElement;
+    return this._parent !== undefined && this._parent instanceof SchemaElement;
```

Comment:
Nit: maybe replacing the first condition by `this.isAttached()` would be ever so slightly more readable.

Reply:
Is the instanceof check unnecessary here?

Reply:
I believe so. The subtlety is that directive can actually be attached to queries 
 and in that case the parent is an `AbstractOperationElement` from `operation.ts`), but while such operation element does link back to a particular schema ultimately (the schema on which the operation applies), that link is "external" and this ensure modifying a direction on an operation isn't considered as a schema modification.